lib/active_model/serializer.rb

```diff
@@ -108,20 +110,50 @@ def self.digest_caller_file(caller_line)
       Digest::MD5.hexdigest(serializer_file_contents)
     end
 
-    def self.get_serializer_for(klass)
+    # Compute the lookup chain for a given serializer class, parent serializer and controller
+    # @param [String] serializer_class_name The class name to lookup
+    # @param [ActiveModel::Serializer] parent_serializer The instance of the parent serializer, if any
+    # @param [ActionController] controller The instance of the calling controller, if any
+    #
+    # @return [Array<String>] The lookup chain
+    #
+    def self.serializer_lookup_chain_for(serializer_class_name, parent_serializer, controller)
+      chain = []
+
+      # Look for a serializer nested inside the current serializer first, if inside a user-defined serializer
+      chain.push("#{self}::#{serializer_class_name}") if self.class != ActiveModel::Serializer
```

Comment:
That kind of check makes me believe it might be time to make `serializer_for` an instance method?